### PR TITLE
new manipulation SetOperationModeStatusAndSetOperatingMode

### PR DESCRIPTION
**Pull request by jannast**
// new manipulation SetOperationModeStatusAndSetOperatingMode

# Checklist

The following aspects have been respected by the author of this pull request, confirmed by both pull request assignee **and** reviewer:

* Changelog update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer
* README update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer

### CODE REVIEW

File: src/t2iapi/combined/combined_requests.proto [hunk 1]
```diff
@@ -57,3 +58,15 @@ message SetAlertActivationAndSetOperatingModeRequest {
...
+Request to set the status of an operation mode of an enum metric with the given handle to the requested value and
+set the @OperatingMode of the operation with the given handle to the requested OperatingMode value.
+*/
+message SetOperationModeStatusAndSetOperatingModeRequest {
+    string metric_descriptor_handle = 1; // handle of the enum metric, which represents the operation mode
+    t2iapi.metric.OperationModeStatus operation_mode_status = 2;  // defines the operation mode status to be applied
+    string operation_descriptor_handle = 3; // the operation descriptor handle for which the OperatingMode value
+                                            // shall be set
+    t2iapi.operation.OperatingMode operating_mode = 4; // OperatingMode value to be set for the operation
+}
```
Comment (ldeichmann) [inline src/t2iapi/combined/combined_requests.proto:72]: Indentation is off
```suggestion
  /*
  Request to set the status of an operation mode of an enum metric with the given handle to the requested value and
  set the @OperatingMode of the operation with the given handle to the requested OperatingMode value.
  */
  message SetOperationModeStatusAndSetOperatingModeRequest {
    string metric_descriptor_handle = 1; // handle of the enum metric, which represents the operation mode
    t2iapi.metric.OperationModeStatus operation_mode_status = 2;  // defines the operation mode status to be applied
    string operation_descriptor_handle = 3; // the operation descriptor handle for which the OperatingMode value
                                            // shall be set
    t2iapi.operation.OperatingMode operating_mode = 4; // OperatingMode value to be set for the operation
  }
```

**Review by ldeichmann — CHANGES_REQUESTED** [review]

**Review by ldeichmann — APPROVED** [review]

File: src/t2iapi/combined/combined_requests.proto [hunk 2]
```diff
@@ -57,3 +58,15 @@ message SetAlertActivationAndSetOperatingModeRequest {
                                             // shall be set
     t2iapi.operation.OperatingMode operating_mode = 4; // the requested OperatingMode value of the operation state
 }
+
+/*
+Request to set the status of an operation mode of an enum metric with the given handle to the requested value and
```
Comment (pstaeck) [inline src/t2iapi/combined/combined_requests.proto:63]: maybe change the wording to sth. like this (same for the rpc):

"Request to set the mode of operation represented by an enum metric with the given handle to the requested value ..."

from my point of view this is clearer

Comment (jannast) [reply src/t2iapi/combined/combined_requests.proto:63]: in the requirement is the wording "operation mode" that's why I think it is better. The similar wording is (was) also used in the function ChangeOperationModeStatusRequest.

Comment (jannast) [reply src/t2iapi/combined/combined_requests.proto]: after review of the new released mpkp standard, the wording should be 'mode of operation'. -> done

File: src/t2iapi/combined/service.proto [hunk 3]
```diff
@@ -51,4 +51,15 @@ service CombinedService {
   rpc SetAlertActivationAndSetOperatingMode (
   t2iapi.combined.SetAlertActivationAndSetOperatingModeRequest)
       returns (BasicResponse);
+
+  /*
+  Request to set the non-exclusive mode of operation represented by an enum metric with the given handle to the
```
Comment (pstaeck) [inline src/t2iapi/combined/service.proto:56]: "Set the ..." instead of "Request to set the ..."

Comment (pstaeck) [reply src/t2iapi/combined/service.proto:56]: After discussion, the wording is not my preferred choice but it's ok for me. We can leave it as it is.

**Review by pstaeck — APPROVED** [review]

**Review by ldeichmann — APPROVED** [review]

**Pull request merged**